js/app.jsx

```diff
@@ -693,12 +693,16 @@ var ModifyTagPopWindow = React.createClass({
       });
 
       var updateTagUriList = {};
+      var updateTagName;
 
       updateTagUriList['tags'] = [];
-      for (var i = 0; i < currentTag['objects'].length; i++) {
-        for (var j = 0; j < updateTagNameList.length; j++) {
-          if (currentTag['objects'][i]['name'] == updateTagNameList[j]) {
+      while (updateTagNameList.length != 0) {
+        updateTagName = updateTagNameList.pop();
+
+        for (var i = 0; i < currentTag['objects'].length; i++) {
```

Comment:
Would it be clearer if you use `Array.prototype.filter()` instead? Something like this:

```
currentTag['objects'].filter(function(x){ return x['name'] == updateTagName})
```